compiler/rustc_parse/src/parser/mod.rs

```diff
@@ -1441,6 +1441,14 @@ impl<'a> Parser<'a> {
             }
         }
 
+        // Recover from `pub let`
+        if self.token.is_keyword(kw::Let) {
```

Comment:
```suggestion
        if self.may_recover() && self.token.is_keyword(kw::Let) {
```

This needs a check whether recovery is allowed, as doing eager recovery here would break the following code:
```rust
macro_rules! uwu {
    ($vis:vis let) => {}
}

uwu! { pub let }
```